monkey/agent_plugins/credentials_collectors/mimikatz/src/plugin.py

```diff
@@ -35,9 +35,10 @@ def __init__(
     def run(self, *, options=None, interrupt=None) -> Sequence[Credentials]:
         logger.info("Attempting to collect windows credentials with pypykatz.")
         windows_credentials = get_windows_creds()
-        logger.info(f"Pypykatz gathered {len(windows_credentials)} credentials.")
+        unique_credentials = list(set(windows_credentials))
+        logger.info(f"Pypykatz gathered {len(unique_credentials)} credentials.")
 
-        collected_credentials = self._to_credentials(windows_credentials)
+        collected_credentials = self._to_credentials(unique_credentials)
```

Comment:
This is fine. The other option is

```
collected_credentials = list(set(self._to_credentials(windows_credentials)))
logger.info(f"Pypykatz gathered {len(collected_credentials)} credentials.")
```

Then you don't meed to modify `WindowsCredentials` at all.

I'll leave the choice to you.

Reply:
That seems to not work for certain test:
<img width="1336" alt="image" src="https://user-images.githubusercontent.com/15820737/235906362-02854672-471b-4a7d-9a3a-6104ae70aacf.png">

Reply:
Switch the order of credentials in `expected_credentials` in the test and try.